datasets/wikicorpus/wikicorpus.py

```diff
@@ -0,0 +1,179 @@
+}
+"""
+
+_DESCRIPTION = """\
+The Wikicorpus is a trilingual corpus (Catalan, Spanish, English) that contains large portions of the Wikipedia (based on a 2006 dump) and has been automatically enriched with linguistic information. In its present version, it contains over 750 million words.
+"""
+
+_HOMEPAGE = "https://www.cs.upc.edu/~nlp/wikicorpus/"
+
+_LICENSE = "GNU Free Documentation License"
+
+_URLs = "https://www.cs.upc.edu/~nlp/wikicorpus/{form}.{language}.tgz"
+
+_LANGUAGES = ["ca", "es", "en"]
+_FORMS = ["raw", "tagged"]
+
+METADATA_PATTERN = re.compile(r'.+id="(?P<id>[^"]+)".+title="(?P<title>[^"]+)".+')
+
+
+class WikicorpusConfig(datasets.BuilderConfig):
+    """ BuilderConfig for Wikicorpus."""
+
+    def __init__(self, form=None, language=None, **kwargs):
+        """
+        Args:
+            form: form of the dataset.
+            language: language of the dataset.
+            **kwargs: keyword arguments forwarded to super.
+        """
+        super().__init__(
+            name=f"{form}_{language}",
+            description=f"Wikicorpus dataset in {form} form and {language} language.",
+            **kwargs,
+        )
+        self.form = form
+        self.language = language
+
+
+class Wikicorpus(datasets.GeneratorBasedBuilder):
+    """Wikicorpus dataset."""
+
+    VERSION = datasets.Version("1.0.0")
+    BUILDER_CONFIG_CLASS = WikicorpusConfig
+    BUILDER_CONFIGS = [WikicorpusConfig(form=form, language=language) for form in _FORMS for language in _LANGUAGES]
+
+    def _info(self):
+        if self.config.form == "raw":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "title": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                }
+            )
+        elif self.config.form == "tagged":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "title": datasets.Value("string"),
+                    "sentence": datasets.Sequence(datasets.Value("string")),
+                    "lemmas": datasets.Sequence(datasets.Value("string")),
+                    "pos_tags": datasets.Sequence(datasets.Value("string")),
```

Comment:
Could you use the `ClassLabel` feature type here ?

You can find an example of usage in the [flores](https://github.com/huggingface/datasets/blob/master/datasets/flores/flores.py) dataset

Reply:
edit : conll, not flores sorry

Reply:
@lhoestq pos_tags in this dataset are not the same as the ones in conll. Here there are **hundreds** (maybe one thousand) of different PoS tag values, like: 
```
'DA0CS0', 'DA0FP0', 'DA0FS0', 'DA0MP0', 'DA0MS0', 'DD0CP0', 'DD0CS0', 'DD0FP0', 'DD0FS0', 'DD0MP0', 'DD0MS0', 
'DE0FS0', 'DE0MP0', 'DE0MS0', 'DI0CN0',...
...
'NC00000', 'NCCN000', 'NCCP000', 'NCCS000', 'NCFN000', 'NCFP000', 'NCFS000', 'NCMN000', 'NCMP000', 'NCMS000',
'NP00000',...
...
'P010P000', 'P010S000', 'P020P000', 'P020S000', 'P0300000', 'PD0CP000', 'PD0CS000', 'PD0FP000', 'PD0FS000', 
'PD0MP000', 'PD0MS000', 'PD0NS000', 'PI0CN000', 'PI0CP000',...
...
'VSIF2P0', 'VSIF2S0', 'VSIF3P0', 'VSIF3S0', 'VSII1P0', 'VSII1S0', 'VSII2P0', 'VSII2S0', 'VSII3P0', 'VSIP1P0', 
'VSIP1S0', 'VSIP2P0', 'VSIP2S0', 'VSIP3P0', 'VSIP3S0', 'VSIS1P0', 'VSIS1S0', 'VSIS2S0', 'VSIS3P0',...
```
Do you think it is a good idea to try to enumerate all of them?

Reply:
See more info here: https://freeling-user-manual.readthedocs.io/en/latest/modules/tagger/#relaxation-labelling-pos-tagger

Reply:
Oh right I didn't know that.
Well then let's keep it this way then and let the users do what they want with the labels